elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -78,7 +108,8 @@ public FilterExpression visitPermissionClass(ExpressionParser.PermissionClassCon
             }
             return ((FilterExpressionCheck) check).getFilterExpression(requestScope);
         } else if (UserCheck.class.isAssignableFrom(check.getClass())) {
-            return USER_CHECK_EXPRESSION;
+            boolean userCheckResult = check.ok(requestScope.getUser());
```

Comment:
This does bypass calling executor, but since we already have the check itself at this point, is it necessary to call the checkUserPermission with executor?